### PR TITLE
chore: Diff-Tool/Staging Directory 

### PR DESCRIPTION
**Pull request by ddixit14**
To run the diff-tool, simple checkout this branch on your local, give any execute permissions needed for the scripts, and run the ./generation/diff_tool.sh.
It will create two text files, `diffSummary.txt` which contains diff failure (diffs found), and `matchSummary.txt` which contains diff success (no diff found).
Below are the files created after a full run ->
[diffSummary.txt](https://github.com/googleapis/google-cloud-java/files/9626515/diffSummary.txt)
[matchSummary.txt](https://github.com/googleapis/google-cloud-java/files/9626529/matchSummary.txt)

### CODE REVIEW

**Comment by ddixit14** [comment]
(Please do not delete this PR, I am using this branch for staging process)

**Comment by suztomo** [comment]
> give any execute permissions needed for the scripts

The user of the script should not do this. You do that when you commit files to repository.

**Comment by suztomo** [comment]
> This branch has conflicts that must be resolved

**Comment by suztomo** [comment]
@ddixit14 Can you clarify the intention of this pull request? Do you intend to merge this pull request once approved? I see ~ 800 pom.xml files are modified here.

**Comment by ddixit14** [comment]
> @ddixit14 Can you clarify the intention of this pull request? Do you intend to merge this pull request once approved? I see ~ 800 pom.xml files are modified here.

No, this should not be merged. I can merge the three diff scripts separately. This branch is entirely to stage directories, and then run the diff_tool to check for differences. Once we are confident that we are good to release from monorepo, I will merge only the three diff scripts. Diff-scripts need a branch which does not have any snapshot versions in it, so a release-please PR branch would be the best candidate going forward.

File: generation/diff_jar.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+# this command generates difference in files for the $artifactName. If diff found, writes to diffSummary.txt
+#no diff found (success) results are written to matchSummary.txt
+
+wget -O mavenFile $1
+
+wget -O sonatypeFile $2
+
+unzip -l mavenFile >>output-1.txt
+#this command checks for filesize differences along with filename diff
+cat output-1.txt | awk 'BEGIN{OFS=":"} {print $4,$1}' | sort >>firstFile.txt
```
Comment (suztomo) [inline generation/diff_jar.sh:12]: How about introducing a flag? Such as "COMPARE_SIZE" environment variable.

File: generation/diff_tool.sh [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+#run this script from google-cloud-java directory
+# google-cloud-java$ ./generation/diff_tool.sh
+# not using set -e because failing diff command is ok
+# was thinking to use staging-directory ID as an argument to this script, but we have different staging ID's for different
+# group ID artifacts so it will complicate the command.
+# latest repositories Id -> comgooglecloud-5570, comgoogleapi-5038, comgoogleanalytics-1052
+# Whenever staging happens, search the logs for the latest-ids and edit them below
+
+set -x
+
```
Comment (suztomo) [inline generation/diff_tool.sh:11]: Add how to get these values.

File: generation/diff_tool.sh [hunk 3]
```diff
@@ -0,0 +1,142 @@
+#!/bin/bash
+#run this script from google-cloud-java directory
+# google-cloud-java$ ./generation/diff_tool.sh
```
Comment (suztomo) [inline generation/diff_tool.sh:3]: Add output of this script.

Comment (suztomo) [reply generation/diff_tool.sh:3]: Explain environmental variable to toggle size check.

File: .kokoro/release/publish_javadoc11.sh [hunk 4]
```diff
@@ -48,16 +41,18 @@ do
   # Proceed if module is not excluded
   if [[ ! "${excluded_modules[*]}" =~ $module ]]; then
     pushd $module
-    # Extract Cloud RAD module name
-    NAME=$(jq -r '.distribution_name' .repo-metadata.json | cut -d ':' -f 2)
```
Comment (suztomo) [inline .kokoro/release/publish_javadoc11.sh:52]: The changes in this .kokoro/release/publish_javadoc11.sh is irrelevant to diff-tool, right?

File: .kokoro/release/common.cfg [hunk 5]
```diff
@@ -9,7 +9,7 @@ build_file: "google-cloud-java/.kokoro/trampoline.sh"
 # Configure the docker image for kokoro-trampoline.
 env_vars: {
   key: "TRAMPOLINE_IMAGE"
-  value: "gcr.io/cloud-devrel-kokoro-resources/java11"
+  value: "gcr.io/cloud-devrel-kokoro-resources/java8"
```
Comment (suztomo) [inline .kokoro/release/common.cfg:12]: Create a pull request that only focuses this change.

**Review by suztomo — APPROVED** [review]

**Review by meltsufin — COMMENTED** [review]
Can we remove all the non `diff_*.sh` files from this PR, and merge this in?

**Pull request closed without merging**